pyomo/opt/base/solvers.py

```diff
@@ -649,6 +648,18 @@ def solve(self, *args, **kwds):
                     assert len(getattr(_model, "._symbol_maps")) == 1
                     delattr(_model, "._symbol_maps")
                     del result._smap_id
+
+                    if self._load_solutions and \
+                       (len(result.solution) == 0):
+                        raise ValueError(
+                            "No solution is available. Try calling "
+                            "solve with the 'load_solutions' keyword "
+                            "set to False. Then, check the statuses "
+                            "on the results object before attempting "
+                            "to load the solution (if one is available). "
+                            "Additionally, one can call solve with the "
+                            "'tee' keyword set to True to view the solver "
+                            "output.")
```

Comment:
Should this be an exception or just an Error logged to the logger?  I could see users just wanting an error (after all, they could be about to check the results object for the solver status anyway).  As we move forward, we should be actively *discouraging* users from requesting the solution in the results object: that is a very slow data structure to generate, and for most situations, a pointless step.  The way this is written, if the user *ever* suspected that the model might fail, then they will be forced to specify `load_solutions=False`...